editor/scene_tree_editor.cpp

```diff
@@ -424,7 +424,9 @@ bool SceneTreeEditor::_add_nodes(Node *p_node, TreeItem *p_parent, bool p_scroll
 		item->set_as_cursor(0);
 	}
 
-	bool keep = (filter.is_subsequence_ofn(String(p_node->get_name())));
+	bool is_kept_by_type = (filter.begins_with("type:") && filter.trim_prefix("type:").is_subsequence_ofn(p_node->get_class())) || (filter.begins_with("t:") && filter.trim_prefix("t:").is_subsequence_ofn(p_node->get_class()));
```

Comment:
IMHO, if `t:` is supported, then better also support `ty:` and `typ:`.

I suggest spliting the search string into two parts separated by `:` , i.e. `[option:]keyword`. When option is specified and is a substring of `type`, filter will type, otherwise do regular searching with the whole search string.

Reply:
OK, will implement it after rebase.

Reply:
I don't really see the point of adding multiple abbreviations of "type", as "t" is enough as a shortcut. If you start typing more, might as well write the whole thing.